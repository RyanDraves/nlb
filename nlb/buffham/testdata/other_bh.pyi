# @generated by Buffham
import dataclasses
from typing import Self

from emb.network.serialize import bh_cobs
from emb.network.transport import transporter
from nlb.buffham import bh

OTHER_CONSTANT = 2


@dataclasses.dataclass
class Ping:
    """This file shares a message name with `sample.bh`
    """

    pong: int

    def serialize(self) -> bytes: ...

    @classmethod
    def deserialize(cls, buffer: bytes) -> tuple[Self, int]: ...

class OtherSerializer(bh_cobs.BhCobs):
    def __init__(self, registry: bh_cobs.Registry | None = None): ...

class OtherNode[Transporter: transporter.TransporterLike](bh.BhNode[OtherSerializer, Transporter]):
    def __init__(self, serializer: OtherSerializer | None = None, transporter: Transporter | None = None): ...

PONG: bh.Transaction[Ping, Ping] = ...
