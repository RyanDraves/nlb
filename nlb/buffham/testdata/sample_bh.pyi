# @generated by Buffham
import dataclasses
import enum
from typing import Self, Type

from emb.network.serialize import bh_cobs
from emb.network.transport import transporter
from nlb.buffham import bh

from nlb.buffham.testdata import other_bh

# This is a constant in the global scope
MY_CONSTANT = 4
CONSTANT_STRING = "Hello, world!"  # constants can have inline comments
# Constants may reference other constants with {brackets}
COMPOSED_CONSTANT = 2 + MY_CONSTANT + other_bh.OTHER_CONSTANT


@dataclasses.dataclass
class Ping:
    """A message comment
    """

    # Add some comments here
    ping: int

    def serialize(self) -> bytes: ...

    @classmethod
    def deserialize(cls, buffer: bytes) -> tuple[Self, int]: ...

@dataclasses.dataclass
class FlashPage:
    """
    A bunch of message comments,
    in a block-like pattern.

    All of these belong to `FlashPage`

    """

    address: int
    # Another field comment
    data: list[int]  # What about some in-line comments for fields?
    # This comment belongs to `read_size`
    read_size: int

    def serialize(self) -> bytes: ...

    @classmethod
    def deserialize(cls, buffer: bytes) -> tuple[Self, int]: ...

@dataclasses.dataclass
class LogMessage:
    message: str

    def serialize(self) -> bytes: ...

    @classmethod
    def deserialize(cls, buffer: bytes) -> tuple[Self, int]: ...

@dataclasses.dataclass
class NestedMessage:
    flag: int
    message: LogMessage
    numbers: list[int]
    pong: Ping
    other_pong: other_bh.Pong

    def serialize(self) -> bytes: ...

    @classmethod
    def deserialize(cls, buffer: bytes) -> tuple[Self, int]: ...

REGISTRY: dict[int, Type[bh.BuffhamLike]] = ...

class SampleSerializer(bh_cobs.BhCobs):
    def __init__(self, registry: bh_cobs.Registry | None = None): ...

class SampleNode[Transporter: transporter.TransporterLike](bh.BhNode[SampleSerializer, Transporter]):
    def __init__(self, serializer: SampleSerializer | None = None, transporter: Transporter | None = None): ...

PING: bh.Transaction[other_bh.Pong, LogMessage] = ...
# Transaction comment
FLASH_PAGE: bh.Transaction[FlashPage, FlashPage] = ...
READ_FLASH_PAGE: bh.Transaction[FlashPage, FlashPage] = ...

class PublishIds(enum.Enum):
    # Publish comment
    LOG_MESSAGE = 4
