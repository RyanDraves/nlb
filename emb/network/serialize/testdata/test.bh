message Foo {
    uint32_t bar;
    string baz;
    list[uint16_t] qux;
}

message Unaligned {
    uint8_t a;
    uint16_t b;
    int32_t c;
    uint64_t d;
    string e;
    list[int64_t] f;
    list[uint8_t] g;
    list[uint16_t] h;
    list[uint32_t] i;
}

message NestedMessage {
    Foo foo;
    uint8_t flag;
}
