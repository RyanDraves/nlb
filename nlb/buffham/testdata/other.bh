# Small file to test imports in `sample.bh`

constant uint8_t other_constant = 2;

message Pong {
    uint8_t pong;
}

transaction pong[Pong, Pong];
