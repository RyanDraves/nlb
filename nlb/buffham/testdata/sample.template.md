# Sample template

Here's a neat sample template where we can generate files that reference Buffham constants, like {{ composed_constant }}.
