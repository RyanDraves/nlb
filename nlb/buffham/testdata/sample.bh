import nlb.buffham.testdata.other;

# This is a constant in the global scope
constant uint8_t my_constant = 4;

constant string constant_string = "Hello, world!";  # constants can have inline comments

# Constants may reference other constants with {brackets}
constant uint16_t composed_constant = 2 + {my_constant} + {nlb.buffham.testdata.other.other_constant};

# A message comment
message Ping {
    # Add some comments here
    uint8_t ping;
}

# This comment isn't associated with anything,
# and will be ignored by the parser

#
# A bunch of message comments,
# in a block-like pattern.
#
# All of these belong to `FlashPage`
#
message FlashPage {
    uint32_t address;
    # Another field comment
    list[uint8_t] data;  # What about some in-line comments for fields?
    # This comment belongs to `read_size`
    uint32_t read_size;
}

message LogMessage {  # In-line message comments are ignored
    string message;
}  # This in-line comment is also ignored

message NestedMessage {
    uint8_t flag;
    LogMessage message;
    list[int32_t] numbers;
    Ping pong;
    nlb.buffham.testdata.other.Ping other_pong;
}

transaction ping[nlb.buffham.testdata.other.Ping, LogMessage];
# Transaction comment
transaction flash_page[FlashPage, FlashPage];
transaction read_flash_page[FlashPage, FlashPage];  # In-line transaction comments are ignored
