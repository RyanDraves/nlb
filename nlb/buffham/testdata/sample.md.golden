# Sample template

Here's a neat sample template where we can generate files that reference Buffham constants, like 2 + 4 + 2.
